### PR TITLE
feat: Add Linux compatibility (reflinks + Linuxbrew bottles + ELF patching)

### PR DESCRIPTION
**Pull request by carlulsoe**
## Summary

This PR adds full Linux support to zerobrew, bringing the same fast package management experience to Linux users. Tested on Raspberry Pi 5 (aarch64) with ~7-14x speedup over Linuxbrew.

### Core Features
- **Linux bottle selection** (`zb_core/src/bottle.rs`): Platform-aware selection for `arm64_linux` and `x86_64_linux` bottles
- **Reflink support** (`zb_io/src/materialize.rs`): FICLONE ioctl for btrfs/XFS copy-on-write, automatic fallback on ext4
- **ELF binary patching** (`zb_io/src/materialize.rs`): Uses `patchelf` to fix RPATH/RUNPATH and interpreter paths

### Testing & Docs
- 142 tests (up from 80) — unit, integration, and edge cases
- `README.md`: Linux installation instructions
- `LINUX_COMPAT.md`: Technical implementation details
- `LINUX_BENCHMARKS.md`: Performance comparison vs Linuxbrew

## Benchmarks (Raspberry Pi 5, aarch64)

| Package | Linuxbrew | zerobrew | Speedup |
|---------|-----------|----------|---------|
| jq | 6.19s | 0.45s | **13.8x** |
| ripgrep | 10.36s | 0.68s | **15.2x** |
| fd | 5.73s | 0.36s | **15.9x** |

Cold: ~7x faster (real download). Warm: ~14x faster (store hit).

## Complex Packages Tested
- ✅ ffmpeg (9 deps) — 2.69s
- ✅ node (17 deps) — 16.22s
- ✅ python@3.12 (7 deps) — 9.47s
- ✅ imagemagick (29 deps) — 21.10s

## Requirements
- `patchelf` for ELF binary patching
- Works on ext4, btrfs, XFS

---

## AI Disclosure 🤖

This PR was developed using AI-assisted coding (Claude via [Clawdbot](https://github.com/clawdbot/clawdbot)).

### Implementation Process

The entire Linux port was built in a ~2 hour session using a human-AI collaboration workflow:

1. **Initial exploration** — AI subagent analyzed the codebase, identified macOS-specific code (APFS clonefile, Mach-O patching, bottle selection)

2. **Parallel implementation** — Multiple subagents worked concurrently:
   - Bottle selection + reflink support
   - ELF binary patching via patchelf
   - Test suite expansion

3. **Code review** — Dedicated review subagent found and fixed 3 critical issues (hardcoded macOS tags, clippy errors, overly aggressive interpreter patching)

4. **Testing & benchmarks** — Built, tested on real hardware (Pi 5), benchmarked against Linuxbrew

5. **Polish** — Clippy fixes, documentation, additional edge case tests

### What the human did
- Architectural decisions (which approach to take)
- Reviewed AI output and requested iterations
- Verified benchmark methodology wasn't "cheating"
- Handled GitHub authentication

### What the AI did
- All code implementation
- Test writing (36 new edge case tests)
- Documentation
- Benchmark execution
- Code review and fixes

This felt like a good collaboration model — human provides direction and taste, AI handles implementation at speed.

---

## Notes

- macOS behavior unchanged
- GitHub Actions CI ready but not included (needs `workflow` token scope) — happy to add in follow-up
- One known issue: imagemagick needed manual `libxml2` install (dep chain incomplete) — investigating fix separately

Happy to iterate on feedback!

### CODE REVIEW

**Comment by carlulsoe** [comment]
## Independent Testing — x86_64 (Bluefin OS)

Tested on my personal laptop to validate the ARM64 benchmarks:

**System:** x86_64, Bluefin OS (Fedora-based immutable), Linuxbrew pre-installed

**Results:**

| Scenario | Linuxbrew | zerobrew | Speedup |
|----------|-----------|----------|---------|
| Cold install | ~7s | ~1s | **~7x** |
| Warm install | ~2s | ~0.1-0.2s | **10-20x** |

Matches the ARM64 results — speedups are consistent across architectures.

The warm install difference really shows the architectural win: brew still extracts and validates, while zb just symlinks from the content-addressable store.

---

*Full disclosure: I contributed this PR using AI-assisted development, but ran these benchmarks independently on my own hardware.*

**Comment by carlulsoe** [comment]
Should be good now. But if you find any issues, ff to let me know. I will be looking out for issues myself :)

File: CODE_REVIEW.md
Comment (lucasgelfond) [inline CODE_REVIEW.md:1]: remove

File: README.md [hunk 1]
```diff
@@ -37,10 +37,38 @@ zb gc                # garbage collect unused store entries
 ## Why is it faster?
 
 - **Content-addressable store**: packages are stored by sha256 hash (at `/opt/zerobrew/store/{sha256}/`). Reinstalls are instant if the store entry exists.
-- **APFS clonefile**: materializing from store uses copy-on-write (zero disk overhead).
+- **APFS clonefile** (macOS) / **reflink** (Linux): materializing from store uses copy-on-write (zero disk overhead).
```
Comment (lucasgelfond) [inline README.md:40]: thanks!

Comment (lucasgelfond) [reply README.md:40]: change above to mac!

File: README.md [hunk 2]
```diff
@@ -37,10 +37,38 @@ zb gc                # garbage collect unused store entries
...
+  # Debian/Ubuntu
+  sudo apt install patchelf
+  
+  # Fedora/RHEL
+  sudo dnf install patchelf
+  
+  # Arch
+  sudo pacman -S patchelf
+  ```
+
```
Comment (lucasgelfond) [inline README.md:61]: pref for these in code vs. in requiremenets - probably add to isntall script?

File: TEST_RESULTS.md
Comment (lucasgelfond) [inline TEST_RESULTS.md:1]: remove this - LLM slop

**Comment by maria-rcks** [comment]
agree with @lucasgelfond, please clean up.

**Comment by lucasgelfond** [comment]
Talked about this a bit internally. We're excited about Linux support, but this PR is not very well though through - lots of complexity, huge, a bunch of leftovers. Would love to see something like this land, but should follow contributor guidelines (on the way); for now, small, isolated, tested changes.
